opentelemetry-etw-metrics/src/exporter/mod.rs

```diff
@@ -39,265 +41,135 @@ impl Debug for MetricsExporter {
     }
 }
 
+fn emit_export_metric_service_request(
+    export_metric_service_request: &ExportMetricsServiceRequest,
+) -> MetricResult<()> {
+    let mut encoding_buffer = Vec::with_capacity(export_metric_service_request.encoded_len());
+
+    export_metric_service_request
+        .encode(&mut encoding_buffer)
+        .map_err(|err| MetricError::Other(err.to_string()))?;
+
+    if (encoding_buffer.len()) > etw::MAX_EVENT_SIZE {
```

Comment:
It would be better to check `if export_metric_service_request.encoded_len() > etw::MAX_EVENT_SIZE` beforehand instead of always trying to encode the data into the `vec` and then checking for the length limit.

Reply:
Good catch. Updated in https://github.com/open-telemetry/opentelemetry-rust-contrib/pull/134/commits/1fe84ff7a94f36539f97407e8f2a3841419bdc83